### PR TITLE
Add Server API reference

### PR DESCRIPTION
**Pull request by lorensr**
Rendered page: https://temporal-documentation-git-api-docs.preview.thundergun.io/references/server-api

Most people don't know these exist, and when I share them, people seem to value/appreciate them, so thought they should be discoverable in the docs.

The client service section we can move to dev guide once dev guide has that content.

Also fixed the CSS so that the link underline in code blocks isn't disrupted! 🥳

### CODE REVIEW

File: docs-src/references/server-api.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+- Edit the JSON in the left pane.
+- Hit `Cmd-Enter` or click the play button to get a response from the server on the right.
+
+![ListWorkflowExecutions](/img/ListWorkflowExecutions.png)
+
+One downside compared to [command line](#with-command-line) is it doesn't show enum names, just numbers like `"task_queue_type": 1`.
+
+![DescribeTaskQueue](/img/DescribeTaskQueue.png)
+
+## HTTP API
```
Comment (cretz) [inline docs-src/references/server-api.md:73]: This section should not be published at all IMO. The HTTP API may not even come to fruition (nobody approves it as is and there are dissenting voices), and we should not say the API exposed by ui-server should be relied on programmatically by users (even with may-have-breaking-changes caveats).

File: docs-src/references/server-api.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+- Hit `Cmd-Enter` or click the play button to get a response from the server on the right.
+
+![ListWorkflowExecutions](/img/ListWorkflowExecutions.png)
+
+One downside compared to [command line](#with-command-line) is it doesn't show enum names, just numbers like `"task_queue_type": 1`.
+
+![DescribeTaskQueue](/img/DescribeTaskQueue.png)
+
+<!-- 
+TODO redo this section when we publish a stable HTTP API
```
Comment (cretz) [inline docs-src/references/server-api.md:82]: Not the biggest fan of keeping commented out code here. If you need in a branch or whatever ok.

File: docs-src/references/server-api.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+title: Temporal Server API
+description: You can interact with Temporal Server via one of its APIs.
+tags:
+  - reference
+---
+
+While it's usually easiest to interact with [Temporal Server](/concepts/what-is-the-temporal-server) via a [Client SDK](/concepts/what-is-a-temporal-client) or the [Temporal CLI](https://docs.temporal.io/cli), you can also use one of its APIs:
+
+- [gRPC API](#grpc-api)
+- [HTTP API](#http-api)
```
Comment (cretz) [inline docs-src/references/server-api.md:12]: This section does not exist

File: docs-src/references/server-api.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+- Install [BloomRPC](https://github.com/bloomrpc/bloomrpc#installation).
+- Open the app
+- Select "Import Paths" button on the top-left and enter the path to the cloned repo: `/path/to/api`
+- Select the "Import protos" + button and select this file:
+
+  `/path/to/api/temporal/api/workflowservice/v1/service.proto`
+
+- A list of methods should appear in the sidebar. Select one.
+- Edit the JSON in the left pane.
+- Hit `Cmd-Enter` or click the play button to get a response from the server on the right.
```
Comment (cretz) [inline docs-src/references/server-api.md:73]: This is a platform specific instruction

**Comment by flossypurse** [comment]
@lorensr - we have talked and think we should go another route with this information. Reach out for details.
I have captured the content in the corresponding Jira task so we can reference it in the future.
Closing this PR.

**Pull request closed without merging**